BrainPortal/app/views/tasks/show.html.erb

```diff
@@ -61,29 +61,20 @@
 
     <!-- FAILURE RECOVERY -->
     <% if CbrainTask::FAILED_STATUS.include?(@task.status) %>
-      <%
-        label = (
-          case @task.status
-          when /Setup/       then "Retry Setting Up"
-          when /Cluster/     then "Retry Processing"
-          when /PostProcess/ then "Retry Post Processing"
-          end
-        )
-      %>
-      <%= link_to label, { :action => 'operation', :operation => 'recover', 'tasklist[]' => @task.id }, :class  => "button menu_button", :method => :post %>
+      <%= link_to 'Retry Failed', { :action => 'operation', :operation => 'recover', 'tasklist[]' => @task.id }, :class  => "button menu_button", :method => :post %>
```

Comment:
I don't understand why:

* you removed the nice labels
* we're still allowing this button to appear for states where recovery is not allowed (e.g. 'Terminated', which is in `CbrainTask::FAILED_STATUS`

I'm a bit grumpy, why can't we get this to work just like I originally designed in back in APRIL 2010, where the button only appeared for the proper states?!?